### PR TITLE
MEN-5525 - feat: added per user access token functionality to the user settings page

### PR DESCRIPTION
**Pull request by mzedel**
this is the UI part for mendersoftware/useradm#297

Ticket: MEN-5525
Changelog: Title
Signed-off-by: Manuel Zedel <manuel.zedel@northern.tech>

### CODE REVIEW

**Review by tranchitella — CHANGES_REQUESTED** [review]
I have a few issues with this PR.

The first problem is that once I generated a new token, and I click on the CLOSE button when the dialog shows me the token's value, if I try creating a new token clicking on "GENERATE A TOKEN" the modal dialog shows again the old token and doesn't allow me to create a new one.

The second issue is that sometimes when refreshing the page (in https://localhost/ui/#/settings/my-profile), the list of tokens doesn't show up. Moving to a different view/page and then back to the "My profile" view doesn't fix the issue. A few full refresh actions, though, fix the problem and the list is back.

The third issue is about the expiration: in the API specs you can set an expiration in seconds (expires_in). I don't know what the UX design included, but would it be possible to have a dropdown with the expiration settings (e.g. 1 hour, 24 hours, 7 days, 30 days, 180 days, 1 year or something like that)? If you have other ideas just go for them.

**Comment by mzedel** [comment]
The first issue you noted should now be fixed...

> The second issue is that sometimes when refreshing the page (in https://localhost/ui/#/settings/my-profile), the list of tokens doesn't show up. Moving to a different view/page and then back to the "My profile" view doesn't fix the issue. A few full refresh actions, though, fix the problem and the list is back.

That I can't reproduce, from what I see the token list is retrieved every time you go to the profile as expected 🤷‍♂️.


> The third issue is about the expiration: in the API specs you can set an expiration in seconds (expires_in). I don't know what the UX design included, but would it be possible to have a dropdown with the expiration settings (e.g. 1 hour, 24 hours, 7 days, 30 days, 180 days, 1 year or something like that)? If you have other ideas just go for them.

This is intended to be added at a later iteration.

File: src/js/actions/userActions.js [hunk 1]
```diff
@@ -595,3 +595,23 @@ export const setHideAnnouncement = (shouldHide, userId) => (dispatch, getState)
...
+    };
+    return Promise.resolve(dispatch({ type: UserConstants.UPDATED_USER, user: updatedUser, userId: user.id }));
+  });
+
+const ONE_YEAR = 31536000;
+
+export const generateToken = name => dispatch =>
+  GeneralApi.post(`${useradmApiUrl}/settings/tokens`, { name, expires_in: ONE_YEAR })
+    .then(({ data: token }) => Promise.all([dispatch(getTokens()), token]))
+    .catch(err => commonErrorHandler(err, `There was an error creatin g the token:`, dispatch));
```
Comment (michaelatmender) [inline src/js/actions/userActions.js:614]: Small typo here - remove space in "creatin g"

**Review by michaelatmender — CHANGES_REQUESTED** [review]
Looks good, with just a couple of small comments from me!
I would also list tokens most recent first, if possible - but is that rather something to change on the API?

File: src/js/actions/userActions.js (hunk 1 above)
Comment (mzedel) [reply src/js/actions/userActions.js]: That should now be fixed 👌

**Comment by mzedel** [comment]
> Looks good, with just a couple of small comments from me! I would also list tokens most recent first, if possible - but is that rather something to change on the API?

Thanks for the review! In this case I think the token sorting is ok to be done in the frontend as we've limited the number of tokens - the tokens should now be sorted 👌.

**Comment by tranchitella** [comment]
> This is intended to be added at a later iteration.

Why?

**Comment by tranchitella** [comment]
As discussed out of band, we won't add the expiration column to the UI.

**Review by michaelatmender — APPROVED** [review]

**Review by tranchitella — APPROVED** [review]

**Pull request merged**